### ISSUE TITLE
Unexpected non-zero values in tf.Variable after gather_nd/reshape window update in eager mode

### ISSUE THREAD

**Issue by Wanzy0209** [tensorflow/tensorflow#127002]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

tf 1.15.0

### Custom code

Yes

### Current behavior?

In eager mode I maintain a sliding-window state variable. `state` is a `tf.Variable` initialized to zeros with shape (4, 2048, 1024). I concatenate 2 new event rows along axis 1 (result shape (4, 2050, 1024)), pass the result through an identity-style `tf.gather_nd` that gathers every row in order, reshape back, and assign the trailing 2048 rows into `state`:

    state.assign(concatenated[:, -2048:, :])

Since only the last 2 rows of the assigned slice contain the new events, `state[:, :-2, :]` should remain exactly zero afterwards. However, `tf.reduce_all(state[:, :-2, :] == 0)` evaluates to False: the leading rows of `state` contain unexpected non-zero values after the update.

Expected behavior: only the last 2 rows of `state` should contain the new events; `state[:, :-2, :]` should stay all zeros.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

tf.compat.v1.enable_eager_execution()

try:
    device_name = "/gpu:0" if tf.config.list_physical_devices('GPU') else "/cpu:0"
except AttributeError:
    device_name = "/cpu:0"
    print("Warning: tf.config.list_physical_devices not found, defaulting to CPU.")

print(f"Running test on device: {device_name}")

with tf.device(device_name):
    def slide_to_the_left_tf(state, new_events, arange, dev_null):
        batch_size = tf.shape(new_events)[0]

        concatenated = tf.concat([state, new_events], axis=1)

        batch_idx = tf.range(batch_size, dtype=tf.int32)[:, tf.newaxis]
        arange_expanded = arange[tf.newaxis, :]

        grid_i = tf.tile(batch_idx, [1, tf.shape(arange)[0]])
        grid_j = tf.tile(arange_expanded, [batch_size, 1])

        indices = tf.stack([
            tf.reshape(grid_i, [-1]),
            tf.reshape(grid_j, [-1])
        ], axis=1)

        gathered = tf.gather_nd(concatenated, indices)
        concatenated = tf.reshape(gathered, [batch_size, tf.shape(arange)[0], tf.shape(concatenated)[2]])

        state.assign(concatenated[:, -2048:, :])

        dev_null.assign(concatenated)

    state = tf.Variable(tf.zeros([4, 2048, 1024], dtype=tf.float32))

    new_events = tf.broadcast_to(
        tf.range(1, 3, dtype=tf.float32)[tf.newaxis, :, tf.newaxis],
        [4, 2, 1024]
    )

    arange = tf.range(2050, dtype=tf.int32)
    dev_null = tf.Variable(tf.zeros([4, 2050, 1024], dtype=tf.float32))

    slide_to_the_left_tf(state, new_events, arange, dev_null)

    is_zero = tf.reduce_all(state[:, :-2, :] == 0)

    print(f"State check (first 2046 rows are zero): {is_zero.numpy()}")

    assert is_zero.numpy(), "Miscompilation/Logic error detected: State contains unexpected non-zero values."
```

### Relevant log output

```shell

```

**Comment by Venkat6871** [comment]
Hi @Wanzy0209, thanks for providing the reproduction.

I tested the same reproduction script with TensorFlow 2.21.0 and also with the latest `tf-nightly` (`2.22.0-dev20260909`). In both cases, I was unable to reproduce the reported behavior.

The final check returns:

```text
State check (first 2046 rows are zero): True
```

So the first 2046 rows of `state` remain zero as expected, and the assertion passes.

Since the issue was reported with TensorFlow 1.15.0, could you please try the same reproduction on a current supported version of TensorFlow (2.21.0 or `tf-nightly`) and let us know if you can still reproduce the issue?

I have also provided the Colab [gist](https://colab.sandbox.google.com/gist/Venkat6871/61c68799745b5ec65a97a8dab625b601/127002_tf_2-21-0-nightly-v.ipynb) with my test results for reference. Please take a look and let us know if you see any differences in your environment.

Thanks!
